test/e2e/prometheus_test.go

```diff
@@ -318,3 +318,61 @@ func remoteWriteCheckMetrics(ctx context.Context, t *testing.T) {
 		)
 	}
 }
+
+func TestBodySizeLimit(t *testing.T) {
+	const (
+		pvcClaimName       = "prometheus-data"
+		storage            = "20Gi"
+		bodySizeLimitSmall = "1MB"
+	)
+
+	cm := f.MustGetConfigMap(t, clusterMonitorConfigMapName, f.Ns)
+	cmBackup := cm.DeepCopy()
+	cmBackup.ObjectMeta.ResourceVersion = ""
+	cmBackup.ObjectMeta.UID = ""
+	cmBackup.ObjectMeta.CreationTimestamp = metav1.Time{}
+
+	restoreConfig := func() {
+		f.MustCreateOrUpdateConfigMap(t, cmBackup)
+	}
+
+	defer restoreConfig()
+
+	f.PrometheusK8sClient.WaitForQueryReturnFloat(
+		t, 5*time.Minute, `sum(increase(prometheus_target_scrapes_exceeded_body_size_limit_total{job="prometheus-k8s"}[5m]))`,
+		func(v float64) error {
+			if v > 0.0 {
+				return fmt.Errorf("expected prometheus_target_scrapes_exceeded_body_size_limit_total does not increase up but got %v increase in last 5 minutes", v)
+			}
+
+			return nil
+		},
+	)
+
+	data := fmt.Sprintf(`prometheusK8s:
+  logLevel: debug
+  enforcedBodySizeLimit: %s
+  volumeClaimTemplate:
+    metadata:
+      name: %s
+    annotations:
+      openshift.io/cluster-monitoring-drop-pvc: "yes"
+    spec:
+      resources:
+        requests:
+          storage: %s
+`, bodySizeLimitSmall, pvcClaimName, storage)
+	f.MustCreateOrUpdateConfigMap(t, configMapWithData(t, data))
+
+	f.PrometheusK8sClient.WaitForQueryReturnFloat(
+		t, 5*time.Minute, `sum(increase(prometheus_target_scrapes_exceeded_body_size_limit_total{job="prometheus-k8s"}[5m]))`,
```

Comment:
If you don't want to add the new WaitForQueryReturnFloat() function, you can also tweak the query so it returns always an integer.

```suggestion
		t, 5*time.Minute, `count(increase(prometheus_target_scrapes_exceeded_body_size_limit_total{job="prometheus-k8s"}[5m] > 0))`,
```

we can also check that the returned value is 2 (prometheus-k8s-0 and prometheus-k8s-1).

Reply:
good idea for rounding the number :D we dont need to create a float version for these functions. will modify the code accordingly.